src/Telemetry/AzureDevOpsTimeline/AzureDevOpsTimeline.cs

```diff
@@ -319,5 +364,56 @@ private static string GetBucket(AugmentedTimelineIssue augIssue)
         {
             return await azureServer.ListBuilds(project, cancellationToken, minDateTime, limit);
         }
+
+        private async Task GetImageNames(List<AugmentedTimelineRecord> records, CancellationToken cancellationToken)
+        {
+            SemaphoreSlim throttleSemaphore = new SemaphoreSlim(50);
+
+            var bag = new ConcurrentBag<Task>();
+            int canceledTasksNumber = 0;
+
+            foreach(var record in records)
+            {
+                if(!string.IsNullOrEmpty(record.Raw.Log?.Url) && record.Raw.Name == "Initialize job")
+                {
+                    var childTask = Task.Run(async () =>
```

Comment:
Instead of `Run()` I would just extract this inside into a method and collect the tasks from calling it?

Reply:
That makes sense to me too. Task.Run is, 99% of the time, not needed and generally just leads strange problems (like the AsyncLocal's not flowing in, so logging affinity can get broken)